odl/trafos/wavelet.py

```diff
@@ -652,21 +982,28 @@ def _call(self, x):
             The length of the array depends on the size of input image to
             be transformed and on the chosen wavelet basis.
         """
-        if x.space.ndim == 1:
-            coeff_list = pywt.wavedec(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 2:
-            coeff_list = pywt.wavedec2(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 3:
-            coeff_dict = wavelet_decomposition3d(x, self.wbasis, self.mode,
-                                                 self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_dict, self.size_list)
-
+        if self._axes is None:
+            if x.space.ndim == 1:
+                coeff_list = pywt.wavedec(x, self.wbasis, self.mode,
+                                          self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 2:
+                coeff_list = pywt.wavedec2(x, self.wbasis, self.mode,
+                                           self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 3:
+                coeff_list = wavelet_decomposition3d(x, self.wbasis, self.mode,
+                                                     self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+
+                return self.range.element(coeff_arr)
+        else:
+            coeff_dict = pywt.dwtn(x, self.wbasis, self.mode, self._axes)
```

Comment:
Just curious: What happens if you use this function for standard axes, i.e. `[1] * self.domain.ndim`? Should give the same result as the three guys above. If that is the case, use this function in all cases, unless it is significantly slower (which is unlikely).